packages/core/src/tracing/idleSpan.ts

```diff
@@ -108,6 +109,33 @@ export function startIdleSpan(startSpanOptions: StartSpanOptions, options: Parti
   const previousActiveSpan = getActiveSpan();
   const span = _startIdleSpan(startSpanOptions);
 
+  function _endSpan(timestamp: number = timestampInSeconds()): void {
+    // Ensure we end with the last span timestamp, if possible
+    const spans = getSpanDescendants(span).filter(child => child !== span);
+
+    // If we have no spans, we just end, nothing else to do here
+    if (!spans.length) {
+      span.end(timestamp);
+      return;
+    }
+
+    const latestEndTimestamp = Math.max(...spans.map(span => spanToJSON(span).timestamp || 0));
+
+    if (latestEndTimestamp <= 0) {
+      span.end(timestamp);
+      return;
+    }
```

Comment:
I feel like we should do this to avoid relay droppage when for some reason all the end timestamps in the span tree are before the start timestamp of the root span:

```suggestion
    if (latestEndTimestamp <= span.start_timestamp_or_whatever_the_field_is_called) {
      span.end(span.start_timestamp_or_whatever_the_field_is_called);
      return;
    }
```

Does that make sense?

Reply:
makes sense I think, I'll update this to be safe!